src/core/actions/creators.ts

```diff
@@ -20,125 +20,283 @@ namespace ActionCreators {
   // fetch action creators
   /**
    * Makes a request for more refinements for given navigation.
-   * @param  {string}                       navigationId - The navigationId for
-   * the navigation to fetch more refinements against.
-   * @return {Actions.FetchMoreRefinements}              - Action with navigationId.
+   * @param  {Actions.Payload.Fetch.MoreRefinements} options - An object with the navigationId for
+   * the navigation to fetch more refinements against and a request object for override.
+   * @return {Actions.FetchMoreRefinements} - Action with { navigationId, request }.
+   */
+  export function fetchMoreRefinements(options: Actions.Payload.Fetch.MoreRefinements): Actions.FetchMoreRefinements;
+  /**
+   * Makes a request for more refinements for given navigation.
+   * @param  {string} navigationId - The navigationId for the navigation to fetch more refinements against.
+   * @return {Actions.FetchMoreRefinements} - Action with { navigationId }.
    */
-  export function fetchMoreRefinements(navigationId: string): Actions.FetchMoreRefinements {
-    return createAction(Actions.FETCH_MORE_REFINEMENTS, navigationId);
+  export function fetchMoreRefinements(navigationId: string): Actions.FetchMoreRefinements;
+  // tslint:disable-next-line typedef
+  export function fetchMoreRefinements(options) {
+    if (typeof options === 'string') {
+      return createAction(Actions.FETCH_MORE_REFINEMENTS, { navigationId: options });
```

Comment:
To reduce code duplication, perhaps it would be better to build an options object and `return createAction` once at the end using the object that we build? Nothing else changes other than that object.